CONTRIBUTING.md

```diff
@@ -7,19 +7,101 @@ We're glad you want to contribute markdownlint! This document will help answer c
+1. Follow the [forking workflow](https://guides.github.com/activities/forking/) steps:
+  1. Fork the project ( <http://github.com/markdownlint/markdownlint/fork> )
+  1. Create your feature branch (`git checkout -b my-new-feature`)
+  1. Commit your changes (`git commit -am 'Add some feature'`)
+  1. Push to the branch (`git push origin my-new-feature`)
+1. Create a [GitHub Pull Request](https://help.github.com/articles/about-pull-requests/) for your change, following all [pull request requirements](#pull-request-requirements) and any instructions in the pull request template
+1. Participate in a [Code Review](#code-review-process) with the project maintainers on the pull request
+
+### Your First Code Contribution
+
+Unsure where to begin contributing to Markdownlint? You can start by looking through these beginner and help-wanted issues:
+
+[Beginner issues](https://github.com/markdownlint/markdownlint/issues?q=is%3Aissue+is%3Aopen+label%3A%22good+first+issue%22+sort%3Acomments-desc) - issues which should only require a few lines of code, and a test or two.
+[Help wanted issues](https://github.com/markdownlint/markdownlint/issues?q=is%3aissue+is%3aopen+label%3a%22help+wanted%22+sort%3Acomments-desc) - issues which should be a bit more involved than beginner issues.
+Both issue lists are sorted by total number of comments. While not perfect, number of comments is a reasonable proxy for impact a given change will have.
+
+### Local Development
+
+**Under Development (Oct 2019)**
+<!--
+* Please try not to mess with the Rakefile, version, or history. If you want to have your own version, or is otherwise necessary, that is fine, but please isolate to its own commit so I can cherry-pick around it.
+* [ ] Run `bundle exec rake default`. It executes all tests and Markdownlint for itself, and generates the documentation
+-->
+
+### Pull Request Requirements
+
+Markdownlint strives to ensure high quality for the project. In order to promote this, we require that all pull requests to meet these specifications:
+
+* **Tests:** To ensure high quality code and protect against future regressions, we require tests for all new/changed functionality in Markdownlint. Test positive and negative scenarios, try to break the new code now.
+* **Green CI Tests:** We use [Travis CI](https://travis-ci.org/markdownlint/markdownlint) to test all pull requests. We require these test runs to succeed on every pull request before being merged.
+
+Before submitting the Pull Request, make sure the following are checked:
+
+* Used the same coding conventions as the rest of the project
+* Wrote [good commit messages](https://chris.beams.io/posts/git-commit/)
+* Commit message starts with `[Fix #issue-number]` if there is a corresponding open GitHub issue
+* Feature branch is up-to-date with `master`, if not - rebase it
+* All [related commits squashed together](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html)
```

Comment:
See comments above.

Reply:
Yeah, this will mirror what the template checklist includes - or it can be excluded and only in the pull request template.

Reply:
Yeah I don't think it's worth it. We're gonna squash them all anyway... and as long as they're not squashign on their side, GH will do a "show since last view" diff that's perfect.